### ISSUE TITLE
simplified callback syntax

### ISSUE THREAD

**Issue by rzimmerman** [rzimmerman/kal#11]
Something like:

```
name = "Steve"
db.findByName name yields user
console.log user.id
```

Instead of:

```
name = "Steve"
db.findByName name, (err, user) ->
  return err if err
  console.log user.id
```

**Comment by rzimmerman** [comment]
The [yield](https://github.com/rzimmerman/kal/tree/yield) branch is tracking this feature.

**Comment by rzimmerman** [comment]
Merged in yield branch at 00c490fb1b2ca3d7437ef982b56ad33d79486d0f

This supports (possibly buggy) the following syntax:

```
wait for x from slowIoFunction()
```

for:
- Basic callbacks
- Nested callbacks
- Callbacks in conditionals (ifs)
- Error handling

But not loops yet

**Comment by rzimmerman** [comment]
Merged in all changes, passing tests. Any further fixes should be logged as new issues (bugs): 1db530a6ecb83e17ccac88b33ea535f1d9c95174
